### PR TITLE
adding links to first references of the vocabulary items and support for collapsing doc strings

### PR DESCRIPTION
**Pull request by sanbrock**
created as a split from #1303

### CODE REVIEW

File: dev_tools/docs/nxdl.py [hunk 1]
```diff
@@ -12,10 +12,15 @@
 from ..globals.errors import NXDLParseError
 from ..globals.nxdl import NXDL_NAMESPACE
 from ..globals.urls import REPO_URL
+from ..utils import nxdl_utils as pynxtools_nxlib
```
Comment (PeterC-DLS) [inline dev_tools/docs/nxdl.py:15]: Can you avoid the use of the alias? Just import and use `get_inherited_nodes`.

File: dev_tools/tests/test_nxdl_utils.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
+"""This is a code that performs several tests on nexus tool
+
+"""
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (PeterC-DLS) [inline dev_tools/tests/test_nxdl_utils.py:9]: Should we mix up licences?

File: dev_tools/tests/test_nxdl_utils.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    node = nexus.get_node_at_nxdl_path("/ENTRY/NXODD_name/float_value", elem=elem)
+    assert node.attrib["type"] == "NX_FLOAT"
+    assert node.attrib["name"] == "float_value"
+
+    node = nexus.get_node_at_nxdl_path(
+        "/ENTRY/NXODD_name/AXISNAME/long_name", elem=elem
+    )
+    assert node.attrib["name"] == "long_name"
+
+    # nxdl_file_path = os.path.join(
```
Comment (PeterC-DLS) [inline dev_tools/tests/test_nxdl_utils.py:68]: Please remove all this commented out code. To preserve it, put it in another branch.

File: dev_tools/utils/nxdl_utils.py [hunk 4]
```diff
@@ -0,0 +1,839 @@
...
+
+    contrib_def_path_glob = Path(nexus_def_path) / "contributed_definitions" / "*.nxdl*"
+
+    files = sorted(glob(app_def_path_glob))
+    for nexus_file in sorted(contrib_def_path_glob):
+        root = get_xml_root(nexus_file)
+        if root.attrib["category"] == "application":
+            files.append(nexus_file)
+
+    return [Path(file).name[:-9] for file in files] + ["NXroot"]
```
Comment (PeterC-DLS) [inline dev_tools/utils/nxdl_utils.py:52]: `[:-9]` assumes file ends with `.nxdl.xml` but that's not only what the glob can match

**Comment by RussBerg** [comment]
@sanbrock as requested on this mornings call I built the docs with this PR and it appears to **not** have broken the code for adding the contributors 👍 .

Incidentally you can build the docs locally with the contributors included if you create a github token then set the environment variable **GH_TOKEN** to that token string and it should see the token is defined and use it to retrieve the contributor information via the github API.

File: dev_tools/docs/nxdl.py (hunk 1 above)
Comment (sanbrock) [reply dev_tools/docs/nxdl.py]: DONE

File: dev_tools/tests/test_nxdl_utils.py (hunk 2 above)
Comment (sanbrock) [reply dev_tools/tests/test_nxdl_utils.py]: DONE

File: dev_tools/tests/test_nxdl_utils.py (hunk 3 above)
Comment (sanbrock) [reply dev_tools/tests/test_nxdl_utils.py]: DONE

File: dev_tools/utils/nxdl_utils.py (hunk 4 above)
Comment (sanbrock) [reply dev_tools/utils/nxdl_utils.py:54]: DONE

**Review by PeterC-DLS — APPROVED** [review]
LGTM

**Pull request merged**